build/tools/Automation/DotNet/DotNetAutomator.cs

```diff
@@ -0,0 +1,274 @@
+                VersionType = versionType,
+            };
+        }
+
+        private void UpdateOryxConstantsForNewVersions(List<DotNetVersion> dotNetVersions, List<ConstantsYamlFile> yamlConstants)
+        {
+            Dictionary<string, ConstantsYamlFile> dotnetYamlConstants = this.GetYamlDotNetConstants(yamlConstants);
+
+            // update dotnetcore sdks and runtimes
+            foreach (var dotNetVersion in dotNetVersions)
+            {
+                string version = dotNetVersion.Version;
+                string sha = dotNetVersion.Sha;
+                string versionType = dotNetVersion.VersionType;
+                string dotNetConstantKey = this.GenerateDotNetConstantKey(dotNetVersion);
+                Console.WriteLine($"[UpdateConstants] version: {version} versionType: {versionType} sha: {sha} dotNetConstantKey: {dotNetConstantKey}");
+
+                if (versionType.Equals(DotNetConstants.SdkName))
+                {
+                    ConstantsYamlFile dotNetYamlConstant = dotnetYamlConstants[DotNetConstants.DotNetSdkKey];
+                    dotNetYamlConstant.Constants[dotNetConstantKey] = version;
+
+                    // update versionsToBuild.txt
+                    this.fileService.UpdateVersionsToBuildTxt(
+                        DotNetConstants.DotNetName,
+                        $"\n{dotNetVersion.Version}, {dotNetVersion.Sha},");
+                }
+                else
+                {
+                    ConstantsYamlFile dotNetYamlConstant = dotnetYamlConstants[DotNetConstants.DotNetRuntimeKey];
+                    dotNetYamlConstant.Constants[dotNetConstantKey] = version;
+
+                    // store SHAs for net-core and aspnet-core
+                    dotNetYamlConstant.Constants[$"{dotNetConstantKey}-sha"] = sha;
+                }
+            }
+
+            var constantsYamlSubPath = Path.Combine("build", Constants.ConstantsYaml);
+            this.yamlFileService.WriteConstantsYamlFile(constantsYamlSubPath, yamlConstants);
+        }
+
+        private Dictionary<string, ConstantsYamlFile> GetYamlDotNetConstants(List<ConstantsYamlFile> yamlContents)
+        {
+            var dotnetConstants = yamlContents.Where(c => c.Name == DotNetConstants.DotNetSdkKey || c.Name == DotNetConstants.DotNetRuntimeKey)
+                                  .ToDictionary(c => c.Name, c => c);
+            return dotnetConstants;
+        }
+
+        private string GenerateDotNetConstantKey(DotNetVersion dotNetVersion)
+        {
+            string[] splitVersion = dotNetVersion.Version.Split('.');
+            string majorVersion = splitVersion[0];
+            string minorVersion = splitVersion[1];
+
+            // prevent duplicate majorMinor where 11.0 and 1.10 both will generate a 110 key.
+            int majorVersionInt = int.Parse(majorVersion);
+            string majorMinor = majorVersionInt < 10 ? $"{majorVersion}{minorVersion}" : $"{majorVersion}_{minorVersion}";
```

Comment:
I may have missed the previous comment resolution, but is there a reason we can't just always use the `{majorVersion}_{minorVersion}` pattern as a key?

Reply:
None that I could see. YAML allows `_` and underscore also is not part of semantic versioning.

Reply:
Side note: I've also updated this for PythonAutomator